packages/react-components/src/overlay/src/usePopup.ts

```diff
@@ -71,7 +93,7 @@ export function usePopup(type, {
         trigger,
         onHide: useEventCallback(event => {
             // Ignore events related to the trigger to prevent double toggle.
-            if (event.relatedTarget !== triggerElement) {
+            if ((event as FocusEvent<HTMLElement>).relatedTarget !== triggerElement) {
                 updateIsOpen(event, false);
             }
         }),
```

Comment:
This code is not shown here, but there is a compilation error 
```
    const canHideOnInteractOutside = useCallback(event => {
        return !isTargetParent(event.target, triggerElement);
    }, [triggerElement]);

    const overlayDismissProps = useOverlayLightDismiss(useCommittedRef(overlayElement), {
        trigger,
        onHide: useEventCallback(event => {
            // Ignore events related to the trigger to prevent double toggle.
            if (event.relatedTarget !== triggerElement) {
            if ((event as FocusEvent<HTMLElement>).relatedTarget !== triggerElement) {
                updateIsOpen(event, false);
            }
        }),
        hideOnEscape: isOpen && hideOnEscape,
        hideOnLeave: isOpen && hideOnLeave,
        hideOnOutsideClick: isOpen && hideOnOutsideClick ? canHideOnInteractOutside : false
    });
 ```
`hideOnOutsideClick` should be a boolean, but canHideOnInteractOutside is a function and not a boolean?